_source/_posts/2022-04-15-jhipster-ionic-spring-boot.adoc

```diff
@@ -0,0 +1,402 @@
+
+// borrowed from https://capacitorjs.com/
+image::{% asset_path 'blog/jhipster-ionic-blueprint/capacitor.png' %}[alt=Capacitor diagram,width=300,role="pull-right m-up-20",link={% asset_path 'blog/jhipster-ionic-blueprint/capacitor.png' %}]
+
+Ionic provides support for the big three web frameworks: Angular, React, and Vue. Once you've written your app with one of these, you can deploy it to a simulator or device with https://capacitorjs.com/[Capacitor]. Capacitor (pictured on the right as the blue layer) provides the runtime for your app to communicate with the native operating system and vise versa.
+
+Ionic's main competitors are native apps built with Swift or Objective-C (for iOS) and Java or Kotlin (for Android). Ionic also competes with React Native, which uses web technologies and translates them to native components.
+
+The Ionic blog has a recent post that https://ionicframework.com/blog/ionic-vs-react-native-performance-comparison/[does a performance comparison between Ionic and React Native]. TL;DR: Both options will give you a high performance app with a truly native look and feel.
+
+**Prerequisites:**
+
+- https://nodejs.org[Node 16]
+- https://sdkman.io/[Java 11]
+- https://docs.docker.com/desktop/#download-and-install[Docker Desktop]
+
+toc::[]
+
+== ✨ Introducing the JHipster Ionic blueprint!
+
+The JHipster project has provided support for generating an Ionic app using a `generator-jhipster-ionic` module for the past several years. As the primary maintainer of this module, I'm proud to announce that this module has been re-written as a blueprint and it's _much_ easier to understand now. The previous module relied on the Ionic CLI, the base Angular https://github.com/ionic-team/starters[starter], a https://github.com/oktadev/ionic-jhipster-starter[Ionic JHipster starter], and custom code to glue it all together. Now, the source code is all contained in one project.
+
+https://github.com/mshima[Marcelo Shima] volunteered to do the conversion and after a couple of months, I'm proud to say the https://github.com/jhipster/generator-jhipster-ionic[JHipster Ionic blueprint] is now available! Here's how to use it:
+
+Create a `ionic-app` directory alongside your JHipster app.
+
+[source,shell]
+----
+- jhipster-app
+- ionic-app
+----
+
+Navigate into it using your terminal. Install Ionic for JHipster and create a new app using `jhipster-ionic`.
+
+[source,shell]
+----
+npm install -g generator-jhipster-ionic@8
+jhipster-ionic
+----
+
+You'll be prompted for the location of your JHipster app, a name for your Ionic app, and then you'll be off to the races!
+
+image::{% asset_path 'blog/jhipster-ionic-blueprint/jhipster-ionic.png' %}[alt=JHipster Ionic prompts,width=800,align=center]
```

Comment:
Should I be concerned about the warnings about relationships? I don't recall seeing these when I generated the JHipster app.